src/navigation-instruction.ts

```diff
@@ -121,11 +133,11 @@ export class NavigationInstruction {
   /**
   * Adds a viewPort instruction.
   */
-  addViewPortInstruction(viewPortName: string, strategy: string, moduleId: string, component: any): any {
-    const config = Object.assign({}, this.lifecycleArgs[1], { currentViewPort: viewPortName });
-    let viewportInstruction = this.viewPortInstructions[viewPortName] = {
-      name: viewPortName,
-      strategy: strategy,
+  addViewPortInstruction(viewPortNameOrInstruction: string, strategy: string, moduleId: string, component: any): ViewPortInstruction {
```

Comment:
why rename viewPortName. It is still typed as a string, and I don't know anywhere that passes in a full instruction, nor is there any handling for that.

Reply:
I was basing this on an attempt of overloading `addViewPortInstruction` and deprecate the current one for static router. Forgot to revert the name. for `strategy`, I wanted to but wasn't sure at first as it was a break change, but again, I think it's semi private and internal, probably fine to convert to keyof